pkg/kubectl/cmd/log.go

```diff
@@ -72,115 +72,144 @@ func selectContainer(pod *api.Pod, in io.Reader, out io.Writer) string {
 	}
 }
 
-type logParams struct {
-	containerName string
+type LogsOptions struct {
+	Client *client.Client
+
+	PodNamespace  string
+	PodName       string
+	ContainerName string
+	Follow        bool
+	Timestamps    bool
+	Interactive   bool
+	Previous      bool
+	LimitBytes    int
+	Tail          int
+	SinceTime     string
+	SinceSeconds  time.Duration
+
+	Out io.Writer
 }
 
 // NewCmdLog creates a new pod log command
 func NewCmdLog(f *cmdutil.Factory, out io.Writer) *cobra.Command {
-	params := &logParams{}
+	o := &LogsOptions{
+		Out:         out,
+		Interactive: true,
```

Comment:
> I'm pretty sure this is to make sure that defaulting (below) is still the same.
> 
> Since doing otherwise is a breaking CLI change, we should leave this here.

@kargakis is right.  The value from the flag is never used (even in master).  I'll preserve the flag registration for compatibility, but mark it as deprecated.